R/object-package.R

```diff
@@ -369,3 +369,107 @@ itemize <- function(header, x) {
     "}\n"
   )
 }
+
+
+package_description_urls <- function(x) {
+
+  # DOIs handling
+  # target: from <doi:XX.XXX> to \doi{XX.XXX} to avoid CRAN Notes, etc.
+  dois <- character()
+
+  # Both structures are valid
+  pattern_doi <- "<(DOI:|doi:)(.*?)>"
```

Comment:
I think this code would be easier to understand if you used `str_replace_all()` with a `replacement` function, something like:

```R
x <- str_replace_all(x, "<(DOI:|doi:)(.*?)>", function(match) {
  match <- str_remove(match, "^<(DOI|doi):|>$")
  match <- URLdecode(match)
  paste0("\\doi{", match, "}")
})
```